### PR TITLE
clean up SDA version of hacker news jobs

### PR DESCRIPTION
**Pull request by sryza**
This gets the asset versions of the SDA jobs working. This involved two things:
* Mirroring a bunch of changes that were made to clean up the non-SDA hacker news jobs
* Making the assets partitioned

### CODE REVIEW

File: python_modules/dagster/dagster/core/execution/context/input.py [hunk 1]
```diff
@@ -231,7 +231,10 @@ def partition_key(self) -> str:
 
     @property
     def has_asset_partitions(self) -> bool:
-        return self.step_context.has_asset_partitions_for_input(self.name)
+        if self._step_context is not None:
+            return self._step_context.has_asset_partitions_for_input(self.name)
+        else:
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/core/execution/context/input.py:236]: nit: 

useless "else"

File: python_modules/dagster/dagster/core/execution/context/output.py [hunk 2]
```diff
@@ -263,7 +260,10 @@ def partition_key(self) -> str:
 
     @property
     def has_asset_partitions(self) -> bool:
-        return self.step_context.has_asset_partitions_for_output(self.name)
+        if self._step_context is not None:
+            return self._step_context.has_asset_partitions_for_output(self.name)
+        else:
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/core/execution/context/output.py:265]: same

**Review by OwenKephart — APPROVED** [review]
nice this looks a lot better overall

File: python_modules/dagster/dagster/core/execution/context/input.py (hunk 1 above)
Comment (sryza) [reply python_modules/dagster/dagster/core/execution/context/input.py:236]: I'm not super opinionated, but my general take is that having an `else` in this kind of situation is clearer: explicitly cleaves the space of possibilities into two disjoint sets.

Maybe it's worth doing some project-level alignment on a convention for this.

**Pull request merged**